### PR TITLE
enable linting in ci pipeline

### PR DESCRIPTION
**Pull request by agbpatro**
# Description

- Linting has been disabled on github actions even though support is present in the project. 
- Identifying current linting issues and fixing them so that we can enable it going forward

## Type of change

Please select all options that apply to this change:

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update

# Checklist:

Confirm you have completed the following steps:

- [x] My code follows the style of this project.
- [x] I have performed a self-review of my code.
- [ ] I have made corresponding updates to the documentation.
- [x] I have added/updated unit tests to cover my changes.
- [ ] I have added/updated integration tests to cover my changes.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -11,5 +14,39 @@ linters:
...
+        arguments:
+          - allowedPackages:
+            - "github.com/onsi/ginkgo"
+            - "github.com/onsi/ginkgo/v2"
+            - "github.com/onsi/gomega"
+      - name: empty-block
+      - name: error-naming
+      - name: error-return
+      - name: error-strings
+      - name: errorf
```
Comment (agbpatro) [inline .golangci.yml:37]: for some reason, if rule is not declared, its disabled by default. I am not sure if there is a to just define disabled rules here

File: config/config.go [hunk 2]
```diff
@@ -168,6 +169,7 @@ func (c *Config) AirbrakeTlsConfig() (*tls.Config, error) {
 			return nil, fmt.Errorf("can't properly load cert pair (%s, %s): %s", certPath, keyPath, err.Error())
 		}
 		tlsConfig.Certificates = []tls.Certificate{cert}
+		// nolint:staticcheck
```
Comment (vmallet) [inline config/config.go:172]: This one should probably have a TODO to fix properly eventually.

File: go.mod [hunk 3]
```diff
@@ -8,7 +8,6 @@ require (
 	github.com/aws/aws-sdk-go v1.44.278
 	github.com/beefsack/go-rate v0.0.0-20220214233405-116f4ca011a0
 	github.com/confluentinc/confluent-kafka-go/v2 v2.3.0
-	github.com/golang/protobuf v1.5.3
```
Comment (vmallet) [inline go.mod:11]: 👍

File: logger/logger.go [hunk 4]
```diff
@@ -79,7 +79,7 @@ func NewColorLogrusLogger(context string) (*Logger, error) {
 	return logger, nil
 }
 
-// Set minimum log level for messages
+// SetLogLevel minimum log level for messages
```
Comment (vmallet) [inline logger/logger.go:82]: `SetLogLevel` sets the ...

File: telemetry/record.go [hunk 5]
```diff
@@ -41,7 +41,7 @@ var (
 			return &protos.VehicleConnectivity{}
 		},
 	}
-	scientificNotationFloatRegex = regexp.MustCompile("^[+-]?(\\d*\\.\\d+|\\d+\\.\\d*)([eE][+-]?\\d+)$")
+	scientificNotationFloatRegex = regexp.MustCompile(`^[+-]?(\d*\\.\d+|\d+\.\d*)([eE][+-]?\d+)$`)
```
Comment (vmallet) [inline telemetry/record.go:44]: Did one `\\` get through the cracks here? We should double check that conversion (but the conversion is a good one :) )

File: telemetry/record.go [hunk 6]
```diff
@@ -117,7 +118,7 @@ func (record *Record) Raw() []byte {
 	return record.RawBytes
 }
 
-// Length gets the records flatbuffer payload byte size
+// LengthRawBytes gets the records flatbuffer payload byte size
```
Comment (vmallet) [inline telemetry/record.go:121]: Might as well `records` -> `record's`

File: .golangci.yml [hunk 7]
```diff
@@ -1,8 +1,11 @@
 ---
 run:
   timeout: 4m
-  skip-dirs:
+
+issues:
+  exclude-dirs:
     - internal
```
Comment (vmallet) [inline .golangci.yml:7]: This was there before, but curious: why are we not linting `internal`? `internal` is just a visibility thing, we should still keep the code to the same standard no?

**Review by vmallet — COMMENTED** [review]
Lots of good stuff in there! See comments; there might be an issue in the regex raw-string conversion.

File: telemetry/record.go (hunk 5 above)
Comment (agbpatro) [reply telemetry/record.go]: fixed. unit tests caught my other mistakes but just not this one

File: .golangci.yml (hunk 7 above)
Comment (agbpatro) [reply .golangci.yml]: actually removed. looks like this directory doesn't exists

**Review by patrickdemers6 — APPROVED** [review]
Thank you, this is awesome!

File: config/config.go (hunk 2 above)
Comment (agbpatro) [reply config/config.go:173]: added a TODO. Will file an issue to fix it in a later PR

Comment (vmallet) [reply config/config.go:173]: :) Technically it should be "remove the lint `bypass`" because we want to add the check back, not remove it again..

**Review by vmallet — APPROVED** [review]

**Pull request merged**